src/liboslcomp/artic.cpp

```diff
@@ -474,17 +472,94 @@ ArticTranspiler::transpile_typecast_expression(ASTtypecast_expression* node)
 void
 ArticTranspiler::transpile_type_constructor(ASTtype_constructor* node)
 {
-    NOT_IMPLEMENTED;
+    std::vector<ASTNode::ref> args = {};
+    ASTNode::ref arg_node          = node->args();
+    while (arg_node) {
+        args.push_back(arg_node);
+        arg_node = arg_node->next();
+    }
+    source.add_source(artic_string(node->typespec()), "{");
+    if (node->typespec().is_triple() && args.size() == 1) {
+        args.push_back(args[0]);
+        args.push_back(args[0]);
+    }
+
+    for (size_t i = 0; i < args.size(); ++i) {
+        source.add_source(get_arg_name(node->typespec(), static_cast<int>(i)), " = ");
+        dispatch_node(args[i]);
+        source.add_source(", ");
+
+    }
+    source.add_source("}");
 }
+
+
 void
 ArticTranspiler::transpile_function_call(ASTfunction_call* node)
 {
-    NOT_IMPLEMENTED;
+    if(node->is_struct_ctr()){
+        auto constructor = ASTtype_constructor(nullptr, node->typespec(), node->args().get());
+        transpile_type_constructor(&constructor);
+    } else {
+        std::vector<ASTNode::ref> args = {};
+        auto arg_node = node->args();
+        while(arg_node){
+            args.push_back(arg_node);
+            arg_node = arg_node->next();
+        }
+        source.add_source(node->opname(), "(");
+        for(auto arg : args){
+            dispatch_node(arg);
+            source.add_source(", ");
+        }
+        source.add_source(")");
+
+
+    }
+
 }
 void
 ArticTranspiler::transpile_literal_node(ASTliteral* node)
 {
-    NOT_IMPLEMENTED;
+    if(node->typespec().is_int()){
+        source.add_source(std::to_string(node->intval()));
+    } else if(node->typespec().is_float()){
+        source.add_source(std::to_string(node->floatval()));
+    } else if(node->typespec().is_string()){
+        add_string_constant(node->strval());
+        source.add_source("Strings::", node->strval());
```

Comment:
?

Reply:
this is for converting constant strings into enums, so that all the functions that just use strings as pseudo-enums (like transforming from one coordinate space to another) without actually having to deal with strings